Hippo.Web/ClientApp/src/components/Product/Products.tsx

```diff
@@ -0,0 +1,462 @@
+import { useEffect, useState, useMemo, useCallback } from "react";
+import { Link, useParams } from "react-router-dom";
+
+import { ReactTable } from "../../Shared/ReactTable";
+import { createColumnHelper } from "@tanstack/react-table";
+import { ProductModel } from "../../types";
+import { authenticatedFetch, parseBadRequest } from "../../util/api";
+import { ShowFor } from "../../Shared/ShowFor";
+import { usePromiseNotification } from "../../util/Notifications";
+import { useConfirmationDialog } from "../../Shared/ConfirmationDialog";
+import { notEmptyOrFalsey } from "../../util/ValueChecks";
+
+const defaultProduct: ProductModel = {
+  id: 0,
+  name: "",
+  category: "Memory",
+  description: "",
+  unitPrice: "0.00",
+  units: "",
+  installments: 1,
+  installmentType: "OneTime",
+  lifeCycle: 60,
+};
+
+export const Products = () => {
+  const [notification, setNotification] = usePromiseNotification();
+  const [editProductModel, setEditProductModel] = useState<ProductModel>({
+    ...defaultProduct,
+  });
+  const [editConfirmationTitle, setEditConfirmationTitle] = useState("");
+  const [deleteConfirmationTitle, setDeleteConfirmationTitle] =
+    useState("Delete Product");
+  const [products, setProducts] = useState<ProductModel[]>();
+  const { cluster } = useParams();
+
+  useEffect(() => {
+    const fetchProducts = async () => {
+      const response = await authenticatedFetch(
+        `/api/${cluster}/product/index`,
+      );
+
+      if (response.ok) {
+        const data = await response.json();
+        setProducts(data);
+      } else {
+        alert("Error fetching products");
+      }
+    };
+
+    fetchProducts();
+  }, [cluster]);
+
+  const [getEditConfirmation] = useConfirmationDialog<ProductModel>(
```

Comment:
i think these should be <FormField>s inside a <Form>. little odd because the submit button in the modal will be outside of the form, but you can use a ref to submit

Reply:
I based these off code elsewhere in hippo. Seems to work as expected.

Reply:
yep, i just mean for a consistent pattern. this is using our regular method of storing a local state and editing it `onChange`, but other places we're using `react-hook-form` to handle that. i can do that when i refactor, since seems to me this is also a form, just inside a modal. @srkirkland what do you think?

Reply:
#395